### PR TITLE
Add custom rspec matchers for cleaner specs

### PR DESCRIPTION
**Pull request by abgoldstein**
This is a great utility! While I was making a pass on the README, it seemed like the documentation would be cleaner if you added some rspec matchers.

### CODE REVIEW

File: spec/matchers/log.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+RSpec::Matchers.define :log do |job_output|
```
Comment (kevin-j-m) [inline spec/matchers/log.rb]: While my example in the README does test for output, I'm not sure how useful this matcher would be in practice.

I would guess (though I'm not sure) that the number of jobs that have output, or more specifically a string response, may be minimal, and the value of testing that may be small.

The intention of the `block_for` method is to provide full access to whatever the job is doing, to run future assertions against it. So, if you had a job that gave all users in the system 100 points, you would test that 1) the job ran 2) that users had 100 more points than they did before. A contrived example for sure, but something like: 

```ruby
before do
  User.first.update_attributes(points: 0)
  Clockwork::Test.run(file: clock_file, max_ticks: 1)
  Clockwork::Test.block_for("give everyone points").call
end

after { Clockwork::Test.clear! }

it "gives a user 100 points" do
  expect(User.first.points).to eq 100
end
```

**Comment by kevin-j-m** [comment]
I really like the `have_run` matcher and would like to include it. I don't believe this will expose the matcher outside of the gem though, as it's defined within `spec/support`.

Adding this in would cause this to require rspec 3 or greater, which could limit the audience, though rspec 3 was released in June of 2014.

On the plus side, it looks like the [rspec documentation](https://www.relishapp.com/rspec/rspec-expectations/docs/custom-matchers/define-matcher-outside-rspec) includes information on how to use these matchers in minitest.

**Comment by abgoldstein** [comment]
Sounds good. We can also add tick and/or time range support for the `have_run` matcher if you end up wanting to include it.

**Comment by kevin-j-m** [comment]
@abgoldstein would you like to take a pass at moving the matcher such that it's available for users of the gem? I'm happy to review again.

**Comment by abgoldstein** [comment]
Sure, I'll take a look at it.

File: spec/matchers/have_run.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+RSpec::Matchers.define :have_run do |job_name|
+  match do |clock_test|
+    @times_run ||= 1
+    clock_test.manager.times_run(job_name) == @times_run
```
Comment (kevin-j-m) [inline spec/matchers/have_run.rb]: I think it'd be more beneficial if, by default, it didn't check if it ran only once, but more than 0 times.

If exactly is chained to it, __then__ check if `clock_test.manager.times_run(job_name) == @times_run`.

File: spec/matchers/have_run.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+RSpec::Matchers.define :have_run do |job_name|
+  match do |clock_test|
```
Comment (kevin-j-m) [inline spec/matchers/have_run.rb]: Will `clock_test` only resolve appropriately if in the test, we have all uses of this in a `describe Clockwork::Test {}` block? 

If that's the case, I'm not sure if that's a valid assumption to make for users of the gem, or a reasonable assertion to enforce to use the matcher.

Please let me know if I'm understanding this correctly.

**Comment by kevin-j-m** [comment]
Closed in favor of #6

**Pull request closed without merging**